Project1/Folder.cpp

```diff
@@ -1,3 +1,97 @@
+#include <string>
+#include <iostream>
+#include <vector>
 #include "Folder.h"
 
 
+
+	bool Folder::dirnamecheck(vector<Folder> tree, vector<string> currentpath, string name)
+	{	
+		currentpath.push_back(name);
+		for (size_t i = 0; i < tree.size(); i++)
+		{
+			if (tree[i].AbsolutePath == currentpath)
+			{	
+
+					cout << "ERROR: Directory or file with that name already exists.\n";
+					return false;			
+			}
+		}
+		return true;
+	}	
+	Folder Folder::mk(vector<Folder> tree, vector<string> currentpath, string name, string type )
+	{
+		Folder tempFolder;
+		tempFolder.name = name;		
+		tempFolder.parentfolder = currentpath.back();
+		tempFolder.type = type;
+		tempFolder.AbsolutePath = currentpath;		
+		tempFolder.AbsolutePath.push_back(name);
+		return tempFolder;
+	}
+	vector<string> Folder::split(string input, char charsplit)
+	{
+		vector<string> command;
+		string temp;
+		for (size_t i = 0; i < input.length() + 1; i++)
+		{
+			if (input[i] == charsplit)
+			{
+				command.push_back(temp);
+				temp = "";
+			}
+			if (input[i] != charsplit)
+			{
+				temp = temp + input[i];
+
+			}
+			if (i == input.length() - 1)
+			{
+
+				command.push_back(temp);
+				temp = "";
+
+
+			}
+
+		}
+		return command;
+
+	}
+	vector<string> Folder::FindAbsPath(vector<Folder> tree, vector<string> currentpath, string command, string type)
```

Comment:
Én biztosan nem választanám az utolsó paramétert string típusúnak. Egy karakterláncot nagyon könnyű elgépelni, és egy ismeretlen tartalommal hogyan fut majd le a függvény? Mit vesz ki az üres tempvectorból? Valamilyen szimbolikus állandó, osztálykonstans, akármi is jobb lenne, mert ha ezek valamelyikét gépelik el, akkor a fordító jelezni fogja tudni a hibát.